### ISSUE TITLE
Keyboard type gets reset every time. 

### ISSUE THREAD

**Issue by Diego-MX** [pqrs-org/Karabiner-Elements#4013]
This issue is most likely caused by a configuration setting, however I can't find out about it.  
So I'm hoping the people here can help me set it up. 

From Karabiner, whenever I open the app I get the following message: 

> *Please select the keyboard type you'd like to use
> The keyboard type selected here will be used in Karabiner-Elements, regardless of the physical keyboard you're using.
> (and more keyboard choices: ANSI, ISO, JIS)...

After selecting ISO (as I have a Latin Americcan keyboard), the program starts correctly and I can set some simple modifications.  

But next time I open it, I get the same prompt, and the previously set modifications will have been erased.  So unfortunately, this makes Karabiner unusable in my computer.  

However the issue doesn't seem to be Karabiner's, but the way my computer has been setup by my IT department.  And I'm hoping I can get some help from you guys.    

The other place where the keyboard type shows up in my PC, is in the Keyboard Setup Assistant: 
Settings > Keyboard > Change Keyboard Type... > Keyboard Setup Assistant > [Continue] > 
Identifying Your Keyboard  
Press the key immediately to the right of the Shift key on the left side of the keyboard...

But my computer won't identify the [> <] key, and therefore I assume it isn't recognizing the ISO keyboard. 

So I do understand this is an OS System, and hopefully I can get help to figure it out and be able to take full advantage of Karabiner.  

Thanks.

**Comment by lyonsden69** [comment]
I have the same issue. Running clean install of Sequoia (albeit with OCLP) and a Logitech MX Keys keyboard.

I setup some key changes. It works. Reboot and that prompt appears and my config is gone (the JSON file has reverted back to its virgin state).

Have you got any further with it?

**Comment by tekezo** [comment]
Karabiner-Elements saves its configuration by writing to a file, but it's possible that system administrators have set the system to delete this file during boot.

On such machines, using the export and import feature for the configuration file might be the best way to restore it after each reboot.
https://karabiner-elements.pqrs.org/docs/manual/operation/export/

**Comment by miinwoo715** [comment]
I was experiencing the exact same issue as you, and I found the solution.

In Karabiner-Elements, when you click the Misc tab and select 'Open config folder (~/.config/karabiner)', the Finder window should open immediately. However, it won't open, which means the karabiner folder is missing.

To solve this issue, follow these steps:

1. Open Finder, then go to the Go menu and select Go to Folder, or press Shift + Command + G.
2. Enter the following path and press Enter: ~/.config
3. Inside the config folder, create a new folder and name it 'karabiner'.
4. You may be prompted to enter your password to complete the creation of the folder. Enter your Mac's password.
5. Reopen Karabiner-Elements, apply your desired settings, and then restart Karabiner-Elements. You should find that the settings you made are now saved.


In conclusion, I solved the issue related to saving by manually creating the karabiner folder. Please try this out!

**Comment by farmin** [comment]
That`s it, thank you so much @miinwoo715. I got a fresh OSX install and could never get Karabiner to work. There was some kind of link named karabiner in the config folder. `~/.config/karabiner -> /Users/username/karabiner.d`. 

By deleting this file and creating a folder instead i got Karabiner working without any problems.

**Comment by ensconced** [comment]
Having the same issue here despite having the ~/.config/karabiner/karabiner.json
